phi/geom/_box.py

```diff
@@ -97,6 +97,22 @@ def approximate_signed_distance(self, location):
         distance = math.abs(location - center) - extent * 0.5
         return math.max(distance, 'vector')
 
+    def shift_positions(self, positions: Tensor, outward: bool = True, shift_amount: float = 0) -> Tensor:
```

Comment:
This method needs documentation and the name is not descriptive enough.
How about `repel` or `push`?

Also it seems, this method does not return the new positions but the delta. Wouldn't most applications require the new positions?